crates/ott-balancer-bin/src/balancer.rs

```diff
@@ -343,12 +345,15 @@ impl BalancerContext {
             match locator.load_epoch().cmp(&load_epoch) {
                 std::cmp::Ordering::Less => {
                     // we already have an older version of this room
+                    self.unload_room(monolith_id, metadata.name.clone()).await?;
                     return Err(anyhow::anyhow!("room already loaded"));
                 }
                 std::cmp::Ordering::Greater => {
                     // we have an newer version of this room, remove it
-                    self.remove_room(&metadata.name, locator.monolith_id())
+                    self.unload_room(locator.monolith_id(), metadata.name.clone())
```

Comment:
difference between unload_room() and remove_room()?

Reply:
`unload_room` sends a message to the monolith that the room should be unloaded

`remove_room` modifies `BalancerContext` to remove the room from the balancer's state but doesn't send a message.